backend/gsr_booking/group_logic.py

```diff
@@ -42,36 +43,74 @@ def book_room(self, gid, rid, room_name, start, end, user, group):
         if not gsr:
             raise APIError(f"Unknown GSR GID {gid}")
 
+        start = datetime.datetime.strptime(start, "%Y-%m-%dT%H:%M:%S%z")
+        end = datetime.datetime.strptime(end, "%Y-%m-%dT%H:%M:%S%z")
+
         if gsr.kind == GSR.KIND_WHARTON:
-            wharton_users = self.get_wharton_users(group)
-            for wharton_user in wharton_users:
-                try:
-                    booking = self.bw.book_room(
-                        gid, rid, room_name, start, end, wharton_user.user, group_book=True
-                    )
-                    reservation = Reservation.objects.create(
-                        start=start, end=end, creator=user, group=group
-                    )
-                    booking.reservation = reservation
-                    booking.save()
-                    break
-                except APIError:
-                    pass
+            users = self.get_wharton_users(group)
+            credit_id = gsr.lid
         else:
-            all_users = self.get_all_users(group)
-            for all_user in all_users:
-                try:
-                    booking = self.bw.book_room(
-                        gid, rid, room_name, start, end, all_user.user, group_book=True
-                    )
-                    reservation = Reservation.objects.create(
-                        start=start, end=end, creator=user, group=group
-                    )
-                    booking.reservation = reservation
-                    booking.save()
+            users = self.get_all_users(group)
+            credit_id = CreditType.LIBCAL.value
+
+        total_credits = sum([self.bw.check_credits(usr.user).get(credit_id, 0) for usr in users])
```

Comment:
Nit: change `usr` to `user`. Might be a bit easier to read.

Reply:
i didn't use user since user was an input to the function... would o_user or other be better?

Reply:
Oh, I see haha. In that case let's leave it as is :)